### ISSUE TITLE
Type error: cannot set property 'next' of undefined; block stacks deleted

### ISSUE THREAD

**Issue by jwzimmer-zz** [LLK/scratch-vm#616]
Reported during Extensions Internal Playtest (6/16/17).

**TL;DR:** Block stacks were deleted (when using speech & WeDo extensions). This has the same "cannot set property 'next' of undefined" error as in https://github.com/LLK/scratch-vm/issues/590, which was fixed, but does not have the "cannot read property 'input' of undefined" error from that case. (If it's preferable to re-open #590 instead of having this separate issue, please feel free to close this one /cc @paulkaplan.)

### Expected Behavior

Scripts should not get deleted.

### Actual Behavior

Reported as:
> all my scripts deleted! actually deleted, not just the thing where the wrong sprite's blocks are pulled up

I had several script stacks in the speech extension scripting area, and maybe some blocks in the WeDo and cat areas. I don't remember exactly what happened, but I noticed something was wrong, so I clicked through the various sprite scripting areas. I was hoping I had encountered the issue where the wrong Sprite's script area is being rendered, but the right sprite's script area is actually still intact (just not visible). However, I saw no blocks in the two other areas, and then two blocks (the wrong blocks) in the speech extension scripting area. The blocks were not connected, and there were no inputs to one of the blocks. It bore little resemblance to what had been contained in that area.

**Note the weird variable block where it's not rendering the dropdown menu.** (This was one of the blocks that was present on the speech extension scripting area when I went back to it after my scripts were deleted.)
<img width="1016" alt="screen shot 2017-06-16 at 5 06 32 pm" src="https://user-images.githubusercontent.com/1320456/27244795-34889976-52b6-11e7-8d22-5fae80220782.png">

### Steps to Reproduce

Unclear.

### Operating System and Browser

_https://ericrosenbaum.github.io/scratch-gui/, Mac OS El Capitan, Chrome 59_

**Comment by paulkaplan** [comment]
This is a general issue when blocks are not kept in sync between scratch-blocks and the vm. We have solved many variations of it, so I'm going to close this.

**Comment by BrightLin** [comment]
Excuse  @paulkaplan  How did you solved it?
```
projects.bundle.js:73 Uncaught TypeError: Cannot set property 'next' of undefined
    at e.value (projects.bundle.js:73)
    at e.value (projects.bundle.js:73)
    at t.value (projects.bundle.js:146)
    at Blockly.WorkspaceSvg.Blockly.Workspace.fireChangeListener (projects.bundle.js:159)
    at Blockly.Events.fireNow_ (projects.bundle.js:159)
```
![delete-var](https://user-images.githubusercontent.com/15828392/76310094-1831e500-6309-11ea-83c8-fd10a202c97b.gif)
